types.cc

```diff
@@ -67,14 +70,26 @@ requires requires {
 sstring
 time_point_to_string(const T& tp)
 {
-    int64_t count = tp.time_since_epoch().count();
-    auto time = boost::posix_time::from_time_t(0) + boost::posix_time::milliseconds(count);
-    constexpr std::string_view units = "milliseconds";
-  try {
-    return boost::posix_time::to_iso_extended_string(time);
-  } catch (const std::exception& e) {
-    return format("{} {} ({})", count, units, e.what());
-  }
+    auto count = tp.time_since_epoch().count();
+    auto d = std::div(int64_t(count), int64_t(1000));
+    std::time_t seconds = d.quot;
+    std::tm tm;
+    if (!gmtime_r(&seconds, &tm)) {
+        return fmt::format("{} milliseconds (out of range)", count);
+    }
+    auto millis = d.rem;
+    if (!millis) {
```

Comment:
Why is this an interesting case? Isn't it easier just not to have this if() and print 0 milliseconds?

Reply:
There's a unit test that fails without that...